<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />
  <!--<include file="$(find pr2_description)/robots/upload_pr2.launch" />-->
  <!--<param name="robot_description" command="$(find xacro)/xacro.py '$(find pr2_description)/robots/pr2.urdf.xacro'" />-->

  <!--Bring up th kinematics node for ltm_simple_arm_control-->
  <include file="$(find pr2_arm_kinematics)/launch/pr2_ik_rarm_node.launch"/>

  <!--<node pkg="ltm_ros" type="ltm_ros_node" name="ltm_ros" output="screen" launch-prefix="gdb -ex run -args">-->
    <node pkg="ltm_ros" type="ltm_ros_node" name="ltm_ros" output="screen" launch-prefix="$(arg launch_prefix)" respawn="false">
      <param name="use_model_file" value="true"/>
      <param name="obs_file" value="$(find ltm_ros)/data/observations.txt"/> 
      <param name="bag_file" value="$(find ltm_ros)/data/observations.bag"/> 

      <!-- This is only for dmodel_bank-->
      <rosparam param="model_files" subst_value="True">
        <!--[$(find ltm)/matlab/models/door.mdl]-->
        <!--[$(find ltm)/matlab/models/ball_joint.mdl]-->
        [$(find ltm)/matlab/models/complex_door.mdl,
         $(find ltm)/matlab/models/complex_door.mdl]
      </rosparam>
      <rosparam param="model_offsets_x">
        [0.0,0.0]
      </rosparam>
      <rosparam param="model_offsets_y">
        [0.0,0.0]
      </rosparam>
      <rosparam param="model_offsets_z">
        [0.0,0.0]
      </rosparam>
<!--
      <rosparam param="model_files" subst_value="True">
        [$(find ltm)/matlab/models/door.mdl,
        $(find ltm)/matlab/models/door_rigid.mdl]
      </rosparam>
      <rosparam param="model_offsets_x">
        [1.65,1.65]
      </rosparam>
      <rosparam param="model_offsets_y">
        [0.0,0.0]
      </rosparam>
      <rosparam param="model_offsets_z">
        [1.0,1.0]
      </rosparam>
      -->


      <param name="fprims_file" value="$(find ltm)/matlab/fprims/sphere_20.fprim"/>
      <!--<param name="fprims_file" value="$(find ltm)/matlab/fprims/linear_4.fprim"/>-->

      <!--
    <param name="model_file" value="$(find ltm)/matlab/models/drawer_small.mdl"/>
    <param name="model_offset_x" value="-0.2"/>
    <param name="model_offset_y" value="0.0"/>
    <param name="model_offset_z" value="0.0"/>
    <param name="fprims_file" value="$(find ltm)/matlab/fprims/linear_4.fprim"/>
    <param name="fprims_file" value="$(find ltm)/matlab/fprims/circle_40.fprim"/>
    -->

    <param name="sim_time_step" value="0.6"/> <!--0.2-->
    <param name="simulation" value="true"/> 
    <param name="reference_frame" value="/base_link"/>
    <param name="full_trajectory_execution" value="true"/>
  </node>

  <node pkg="tf" type="static_transform_publisher" name="map_broadcaster" args="0 0 0 0 0 0 1 /base_link /map 1000" />

</launch>
